### PR TITLE
[DS-3967] 7x - Migrate Dockerfile to DSpace/DSpace

### PR DESCRIPTION
**Pull request by terrywbrady**
https://jira.duraspace.org/browse/DS-3967

Derived from https://github.com/DSpace/DSpace/pull/2134 with changes to the webapps symlinks

### CODE REVIEW

File: docker.local.cfg [hunk 1]
```diff
@@ -0,0 +1,4 @@
+dspace.dir = /dspace
```
Comment (tdonohue) [inline docker.local.cfg]: As noted in today's meeting, let's move this to `[src]/src/main/docker/local.cfg`

Again, like in 6.x, let's add in a few comments at the top of the file to describe its usage (used by Docker / DockerFile)

**Comment by ppmdo** [comment]
Tested build and compose on Fedora Linux 27. Works without problems. Used the following docker-compose file: https://github.com/DSpace-Labs/DSpace-Docker-Images/tree/master/docker-compose-files/dspace7-compose.

**Comment by ppmdo** [comment]
Tested build and compose on Fedora Linux 27. Works without problem. Used the following compose file: https://github.com/DSpace-Labs/DSpace-Docker-Images/tree/master/docker-compose-files/dspace7-compose

**Comment by ppmdo** [comment]
Tested build and compose on CentOS 7. No problems. Same compose file as above.

**Review by tdonohue — APPROVED** [review]
👍 LGTM.

**Pull request merged**